cellengine/resources/experiment.py

```diff
@@ -28,12 +34,108 @@
+            mm_field=fields.DateTime(),
+        ),
+        default=ReadOnly(),
+    )  # type: ignore
+    updated: datetime = field(
+        metadata=config(
+            encoder=datetime_to_timestamp,
+            decoder=timestamp_to_datetime,
+            mm_field=fields.DateTime(),
+        ),
+        default=ReadOnly(),
+    )  # type: ignore
+    _active_comp: Optional[Union[int, str]] = field(
+        default=None, metadata=config(field_name="activeCompensation")
+    )
+    deleted: Optional[datetime] = field(
+        default=None,
+        metadata=config(
+            encoder=lambda t: datetime_to_timestamp(t) if t else t,
+            decoder=lambda t: timestamp_to_datetime(t) if t else t,
+            mm_field=fields.DateTime(),
+        ),
+    )
+    data: Optional[Dict[Any, Any]] = field(
+        default=ReadOnly(optional=True)
+    )  # type: ignore
+    deep_updated: datetime = field(
+        metadata=config(
+            encoder=datetime_to_timestamp,
+            decoder=timestamp_to_datetime,
+            mm_field=fields.DateTime(),
+        ),
+        default=ReadOnly(optional=True),
+    )  # type: ignore
+    analysis_source_experiment: Optional[str] = field(
+        default=ReadOnly(optional=True)
+    )  # type: ignore
+    clone_source_experiment: Optional[str] = field(
+        default=ReadOnly(optional=True)
+    )  # type: ignore
+    locked: bool = field(default=ReadOnly())  # type: ignore
+    permissions: List[Dict[Any, Any]] = field(default=ReadOnly())  # type: ignore
+    retention_policy: Dict[str, Any] = field(default=ReadOnly())  # type: ignore
+    revision_source_experiment: Optional[str] = field(
+        default=ReadOnly(optional=True)
+    )  # type: ignore
+    _id: str = field(
+        metadata=config(field_name="_id"), default=ReadOnly()
+    )  # type: ignore
+    revisions: List[Dict[str, Any]] = field(default=ReadOnly())  # type: ignore
+
+    def __repr__(self):
+        return f"Experiment(_id='{self._id}', name='{self.name}')"
+
+    @property
+    def comments(self):
+        """Get comments for experiment.
 
+        Defaults to overwrite; append new comments with
+        experiment.comments.append(dict) with the form:
+        dict = {"insert": "some text",
+        "attributes": {"bold": False, "italic": False, "underline": False}}.
+        """
+        comments = self._comments
+        if type(comments) is not CommentList:
+            self._comments = CommentList(comments)
+        return comments
+
+    @comments.setter
+    def comments(self, comments: Dict[str, Any]):
+        comment = comments.get("insert")
+        if comment:
+            if comment.endswith("\n") is False:
+                comments.update(insert=comment + "\n")
+            self._comments = comments  # type: ignore
 
-class Experiment(_Experiment):
     @staticmethod
-    def get(_id: str = None, name: str = None) -> Experiment:
+    def get(_id: str = None, name: str = None):
```

Comment:
Why remove the type?

Reply:
mm, I think I removed it because I changed the return type to `Union[Dict, Experiment]` before, and then didn't change it back. Thanks